Modules/Core/Common/test/itkCommonTypeTraitsGTest.cxx

```diff
@@ -84,3 +84,30 @@ TEST(CommonTypeTraits, ContinuousIndexIsPOD) {
   EXPECT_EQ(std::is_trivial<T>::value, true);
   EXPECT_EQ(std::is_standard_layout<T>::value, true);
 }
+
+/************ FixedArray: noexcept move checks *************/
+
+/* Dummy class without noexcept move contructors. */
+struct MightThrowAtMove
+{
+    MightThrowAtMove() {}
+    MightThrowAtMove(MightThrowAtMove&&) {}
+    MightThrowAtMove(const MightThrowAtMove&) {}
+};
+
+/* Check that FixedArray works as an agregate class.
+ * Move constructors are noexcept when the contained class has a noexcept move constructor.
+ * And when the contained class might throw at move, the qualifer automatically
+ * propagates it to FixedArray.
+ * This is the same behaviour than std::array. It proves there is no need to
+ * add noexcept (ITK_NOEXCEPT) to the container. This would disallow using it
+ * with might-throw-objects.
+ */
+TEST(CommonTypeTraits, FixedArrayIsNoExceptMovable) {
+  using TInt = itk::FixedArray<int, 3>;
+  EXPECT_EQ(std::is_nothrow_move_constructible<TInt>::value, true);
```

Comment:
Personally I would in general prefer to test at compile-time, instead of at runtime, whenever possible: using `static_assert` instead of any runtime `EXPECT` or `ASSERT` macro. (But we already discussed this at #381.)

Reply:
Having a test framework, I personally prefer this run-time check. Thanks for pointing the compile-time alternative though.